From 2035911a35c540c3f1280c21ddd1dd3514a8ad4a Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Thu, 8 Sep 2016 15:37:04 -0400
Subject: [PATCH 1639/1722] drm/amdgpu: Disable certain CG on Carrizo to
 workaround KFD suspend/resume

After suspend/resume, kfdtest may cause hard hang on Carrizo, which
is caused by certain clock gating feature. We choose to work around the
problem by disabling it.

Change-Id: Ifb358cb96814819e8e8c7a2c59671e48e75b900b
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/vi.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/vi.c b/drivers/gpu/drm/amd/amdgpu/vi.c
index c15734f..ee32f38 100644
--- a/drivers/gpu/drm/amd/amdgpu/vi.c
+++ b/drivers/gpu/drm/amd/amdgpu/vi.c
@@ -974,9 +974,8 @@ static int vi_common_early_init(void *handle)
 			AMD_CG_SUPPORT_GFX_RLC_LS |
 			AMD_CG_SUPPORT_GFX_CP_LS |
 			AMD_CG_SUPPORT_GFX_CGTS |
-			AMD_CG_SUPPORT_GFX_MGLS |
 			AMD_CG_SUPPORT_GFX_CGTS_LS |
-			AMD_CG_SUPPORT_GFX_CGCG |
+			/*AMD_CG_SUPPORT_GFX_CGCG |*/
 			AMD_CG_SUPPORT_GFX_CGLS |
 			AMD_CG_SUPPORT_BIF_LS |
 			AMD_CG_SUPPORT_HDP_MGCG |
-- 
2.7.4

